web/modules/custom/dpl_opening_hours/src/Controller/OpeningHoursEditorController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\dpl_opening_hours\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\node\NodeInterface;
+use Drupal\taxonomy\TermInterface;
+
+/**
+ * Defines OpeningHoursEditorController class.
+ */
+class OpeningHoursEditorController extends ControllerBase {
+
+  /**
+   * Display the opening hours app.
+   *
+   * @param int $node
+   *   The node ID.
+   *
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function content(int $node): array {
+    // Load the node entity.
+    $nodeStorage = $this->entityTypeManager()->getStorage('node');
+    $nodeEntity = $nodeStorage->load($node);
+
+    if (!$nodeEntity instanceof NodeInterface || $nodeEntity->getType() !== 'branch') {
+      return [];
+    }
+
+    // Load taxonomy terms for opening_hours_categories.
+    $taxonomyStorage = $this->entityTypeManager()->getStorage('taxonomy_term');
+    $openingHoursCategories = $taxonomyStorage->loadByProperties([
+      "vid" => "opening_hours_categories",
+    ]);
+
+    if (!is_iterable($openingHoursCategories)) {
+      return [];
+    }
+
+    return [
+      '#theme' => 'dpl_react_app',
+      '#name' => 'opening-hours-editor',
+      '#data' => [
+        'opening-hours-editor-categories-config' => $this->buildOpeningHoursCategoriesArray($openingHoursCategories),
+        'opening-hours-branch-id-config' => $nodeEntity->id(),
+        'opening-hours-remove-event-button-text' => $this->t('Remove event'),
+      ],
+    ];
+  }
+
+  /**
+   * Build the opening hours categories array.
+   *
+   * @param \Drupal\taxonomy\TermInterface[] $openingHoursCategories
+   *   Array of opening hours categories.
+   *
+   * @return mixed[]
+   *   Array of category details.
+   */
+  private function buildOpeningHoursCategoriesArray($openingHoursCategories): array {
+    $categoriesArray = [];
+    if (!is_iterable($openingHoursCategories)) {
+      return $categoriesArray;
+    }
+    foreach ($openingHoursCategories as $category) {
+      $categoriesArray[] = [
+        'title' => $category->getName(),
+        'color' => $this->getOpeningHoursColor($category),
+      ];
+    }
+    return $categoriesArray;
+  }
+
+  /**
+   * Get the color for the opening hours category.
+   *
+   * This method retrieves the color specified in the
+   * 'field_opening_hours_color' of a taxonomy term.
+   *
+   * @param \Drupal\taxonomy\TermInterface $category
+   *   The taxonomy term for the opening hours category.
+   *
+   * @return string
+   *   The opening hours color.
+   */
+  private function getOpeningHoursColor(TermInterface $category): string {
+    /** @var \Drupal\Core\Field\FieldItemInterface $firstItem */
+    $firstItem = $category->get('field_opening_hours_color')->first();
+    /** @var \Drupal\Core\Field\FieldItemInterface $firstItem */
```

Comment:
```suggestion
    /** @var \Drupal\Core\Field\FieldItemInterface $firstItem */
    $firstItem = $category->get('field_opening_hours_color')->first();
```

This line duplicates what is already on L90. I do not see a reason to add it twice.